src/ajax.js

```diff
@@ -270,7 +270,23 @@ function ajaxConvert( s, response, jqXHR, isSuccess ) {
 		}
 	}
 
-	return { state: "success", data: response };
+	if ( jQuery.isThenable(response) ) {
+		return response.then(
+			function(response) {
+				this.responseJSON = response;
+				return { state: "success", data: response };
+			}.bind(jqXHR),
+			function(e) {
+				if ( this[ "throws" ] ) {
+					throw e;
```

Comment:
Addressed. :) It was working as I expected it to because of my unfamiliarity with sync vs. async things inside of `jQuery.Deferred`.